cpp/dolfinx/fem/Function.h

```diff
@@ -346,19 +328,16 @@ class Function
     std::vector<std::int32_t> cells_expr;
     cells_expr.reserve(num_cells);
     // Get mesh and check if mesh is shared
-    if (auto mesh_v = _function_space->mesh();
-        expr_mesh.topology() == mesh_v->topology())
+    if (auto mesh_v = _function_space->mesh(); &expr_mesh == &(*mesh_v))
```

Comment:
`&(*mesh_v)` looks complicated. Are you just trying to check that the two meshes point to the same thing?

Reply:
Yes, that is what I am trying to do. As we do not have an equality operator for mesh, this was the best thing I could think of.

Reply:
If both are shared pointers, `expr_mesh == mesh_v` should work.

Reply:
They are not both shared pointers, as `expr_mesh` is a reference: `const dolfinx::mesh::Mesh<geometry_type>& expr_mesh,`